codegen/cmd_test.go

```diff
@@ -1804,6 +1802,81 @@ roleRef:
 			[]v1.EnvVar{{Name: "FOO", ValueFrom: &v1.EnvVarSource{SecretKeyRef: &v1.SecretKeySelector{LocalObjectReference: v1.LocalObjectReference{Name: "bar"}, Key: "baz"}}}}),
 	)
 
+	DescribeTable("rendering service ports",
+		func(portName string) {
+			cmd := &Command{
+				Chart: &Chart{
+					Operators: []Operator{
+						{
+							Name: "painter",
+							Deployment: Deployment{
+								Container: Container{
+									Image: Image{
+										Tag:        "v0.0.0",
+										Repository: "painter",
+										Registry:   "quay.io/solo-io",
+										PullPolicy: "IfNotPresent",
+									},
+								},
+							},
+							Service: Service{
+								Ports: []ServicePort{{
+									Name:        portName,
+									DefaultPort: 9900,
+								}},
+							},
+						},
+					},
+
+					Values: nil,
+					Data: Data{
+						ApiVersion:  "v1",
+						Description: "",
+						Name:        "Painting Operator",
+						Version:     "v0.0.1",
+						Home:        "https://docs.solo.io/skv2/latest",
+						Sources: []string{
+							"https://github.com/solo-io/skv2",
+						},
+					},
+				},
+
+				ManifestRoot: "codegen/test/chart-svcport",
+			}
+
+			err := cmd.Execute()
+			Expect(err).NotTo(HaveOccurred())
+
+			helmValues := map[string]interface{}{}
+
+			renderedManifests := helmTemplate("codegen/test/chart-svcport", helmValues)
+
+			var renderedSvc *v1.Service
+			decoder := kubeyaml.NewYAMLOrJSONDecoder(bytes.NewBuffer(renderedManifests), 4096)
+			for {
+				obj := &unstructured.Unstructured{}
+				err := decoder.Decode(obj)
+				if err != nil {
+					break
+				}
+				if obj.GetName() != "painter" || obj.GetKind() != "Service" {
+					continue
+				}
+
+				bytes, err := obj.MarshalJSON()
+				Expect(err).NotTo(HaveOccurred())
+				renderedSvc = &v1.Service{}
+				err = json.Unmarshal(bytes, renderedSvc)
+				Expect(err).NotTo(HaveOccurred())
+				break
+			}
+			Expect(renderedSvc).NotTo(BeNil())
+			Expect(renderedSvc.Spec.Ports[0].Name).To(Equal(portName))
+		},
+		Entry("port name without hyphen", "foo"),
```

Comment:
might be overkill but is there value in testing that the value can be set by values (as well as the default value)?

Reply:
Don't think that's necessary because Helm allows overriding properties that have an hyphen, e.g. ` --set glooAgent.ports.grpc-runtimeapi=100`. We'd be testing Helm vs our templating, so seems like an overkill. I tested this manually though.